Feature/setup filter test data

nshumoogum opened the pull request:
### What

Rewrite of filter tests to teardown and setup test data, instead of removing all data at the beginning of running test suite. Requires https://github.com/ONSdigital/dp-filter-api/pull/30 to be merged

### How to review

Run filter API tests by doing the following:

1) Run dataset API and filter API
2) cd dp-api-tests/filterAPI
3) go test ./...

Check logic in tests

### Who can review

Team A, B or @pathimanoj

pathimanoj committed:
initial import api tests

pathimanoj committed:
Added import api and filter api tests

pathimanoj committed:
Code list api tests

nshumoogum committed:
Update service to use environment variables and the ability to setup and
cleanup test data

pathimanoj committed:
Initial tests for dataset api

Includes mgo dependency installed in vendor package and update to
mongo setup and teardown functions.

nshumoogum committed:
Extend dataset api tests coverage for all GET endpoints

Also includes an update to both the init function and mongo package.

nshumoogum committed:
Update initialise functions across API test packages

nshumoogum committed:
Recreating variables wrongly removed in earlier commit

nshumoogum committed:
Update tests for many GET endpoints for the dataset API

Test coverage for th following GET endpoints:
* `/datasets`
* `/datasets/<id>`
* `/datasets/<id>/editions`
* `/datasets/<id>/editions/<edition>`
* `/datasets/<id>/editions/<edition>/versions`
* `/datasets/<id>/editions/<edition>/versions/<version>`
* `/instances`
* `/instances/<id>`

pathimanoj committed:
EXtend dataset API tests (covers PUT & POST endpoints)

Also update `get_editions_test`

Crispioso committed:
Test cases for state filters on instances endpoint

Matt-Guest committed:
Setup tests and small fixes to some tests

Matt-Guest committed:
Moved filter api test into new directory

nshumoogum committed:
Extend `get_instances` test for filtering a list by an invalid state

nshumoogum committed:
Handle errors in dataset API tests

nshumoogum committed:
Merge branch 'feature/filter-api' into dataset-api-tests

nshumoogum committed:
Update README.md

nshumoogum committed:
Update filter API tests, setup and teardown test data

nshumoogum committed:
Rename filterAPI test files

Matt-Guest commented on filterAPI/delete_dimension_option_test.go:
```diff
@@ -0,0 +1,142 @@
+		os.Exit(1)
+	}
+
+	if err := setupInstance(); err != nil {
```
Not all tests need an instance in mongo

Matt-Guest commented on filterAPI/delete_dimension_option_test.go:
```diff
@@ -0,0 +1,142 @@
+			os.Exit(1)
+		}
+
+		// TODO Reinstate commented out code below once API has been updated
```
Is this a bug which has been raised?

Matt-Guest commented on filterAPI/delete_dimension_option_test.go:
```diff
@@ -0,0 +1,142 @@
+		os.Exit(1)
+	}
+
+	if err := mongo.Teardown(database, collection, "_id", filterID); err != nil {
```
Can you also log out the err here?

Matt-Guest commented on filterAPI/initialise.go:
```diff
@@ -0,0 +1,40 @@
+	database    = "filters"
+	collection  = "filters"
+	instanceID  = "789"
+	filterID    = "123"
```
Could we generate the instanceID and filterID when the tests run? So if two tests run against the same database they won't override each other.

Matt-Guest commented on filterAPI/post_dimension_test.go:
```diff
@@ -0,0 +1,121 @@
+	}
+}
+
+// func TestFailureToPostDimension(t *testing.T) {
```
Is this test needed?

nshumoogum replied:
Yes, it shouldn't be commented out.

nshumoogum committed:
Update filter API test - to check instance id error

nshumoogum committed:
Generate uuids for filter API tests

This allows tests to run in parallel without breaking.

Also uncommented out test, log all errors and remove setup of instances
where it is not needed.

nshumoogum committed:
Add go.uuid dependency

Matt-Guest reviewed (approved):
Looks great and all the tests passed on my mac 👍
